data-model.d.ts

```diff
@@ -1,6 +1,18 @@
 /* Data Model Interfaces */
 /* https://rdf.js.org/data-model-spec/ */
 
+/* RDF Modes allow for future expansion into other modes, e.g., "easierRDF" */
+
+/* RDF 1.1 */
+export type RdfMode_11 = 'rdf-1.1';
+
+/* RDF* (rdf-star) */
+export type RdfMode_star = 'rdf-star';
+
+/* Modes officially supported/allowed by RDFJS */
+export type AllowedRdfMode = RdfMode_11 | RdfMode_star;
```

Comment:
Wondering if we want to add generalized RDF here as well?
This was the original motivation for adding the `BaseQuad`.